lib/_processQueue.ts

```diff
@@ -6,37 +6,28 @@
  */
 import { isFunction } from "./utils";
 
-type QueueItemCallback = (err: any, res: any) => void;
-type QueueItem = [string, any, QueueItemCallback?];
-
 export function processQueue(globalObject) {
   // Set pointer which allows renaming of the script
   const pointer = globalObject["AlgoliaAnalyticsObject"] as string;
 
   // Check if there is a queue
   if (pointer) {
-    const queue: QueueItem[] = globalObject[pointer].queue || [];
+    const queue: IArguments[] = globalObject[pointer].queue || [];
 
     // Loop queue and execute functions in the queue
-    queue.forEach(([functionName, functionArguments, functionCallback]) => {
+    queue.forEach((args: IArguments) => {
+      const [functionName, ...functionArguments] = [].slice.call(args);
```

Comment:
I think this could be simplified in:

```ts
queue.forEach(([functionName, ...functionArguments]: IArguments) => {
  if (functionName && isFunction((this as any)[functionName])) {
    this[functionName](...functionArguments);
  }
});
```

Reply:
That would have been possible if the queue items were Arrays but they're Arguments instead.
See how typescripts [destructures this](https://www.typescriptlang.org/play/index.html#src=function%20b(%5Bfirst%2C%20...rest%5D)%20%7B%0D%0A%20%20%20%20console.log(first%2C%20rest)%3B%0D%0A%7D%0D%0A%0D%0Ab(2%2C%203)): it's using splice which is not an Argument method.

Reply:
Oh, I see, missed that queue was an array of Arguments.

Reply:
(But I am very concerned that TypeScript did not trigger any error during my tests)

Reply:
That's because we compile typescript code to with `target: es6`.
If you change target to `es5` than the error is picked us.

cf. on [typescript playgroup](https://typescript-play.js.org/?allowUnusedLabels=true&downlevelIteration=true&noStrictGenericChecks=true&noUnusedLocals=true&noUnusedParameters=true&esModuleInterop=true&preserveConstEnums=true&removeComments=true&skipLibCheck=true&emitDecoratorMetadata=true&target=2#code/GYVwdgxgLglg9mABAcwBQG1gwE4GcoA0iAdKdgKb4C6AXIgJICC2yIAtuWFLgJSIDeAKEQjEEBLjgAbcsSlw0WPIUQV8PANyCAvoL2hIsBImCo+Q0SlQBDFu07ceOoA) you can try switching target.

I think one way we can  immediately fix this is by changing the target to es5. Compilation can anticipate more errors if make the target as close as possible from the target env.

One thing we can further investigate is getting rid of bublé. Not sure what's the benefit in having it.

Reply:
Thanks for the explanation. 

This is a bit out of scope with this PR but yes I think it would be nice to fix this so we do not miss some potential errors.